lib/core/storage.js

```diff
@@ -179,9 +179,24 @@ export default class Storage {
     if (isUnset(value)) {
       Cookies.remove(_key, _options)
     } else {
-      Cookies.set(_key, value, _options)
-    }
 
+      // Support server set cookies
+      if (process.server) {
+        const existingCookieArray = this.ctx.res.getHeader('Set-Cookie')
+
+        // Convert cookie array to cookie string
+        const cookies = existingCookieArray ? parseCookie(existingCookieArray.join('; ')) : {}
+
+        // Set or update the cookie
+        cookies[_key] = value
+
+        // Convert back to array and set the header
+        const newCookieArray = Object.keys(cookies).map(key => serializeCookie(key, cookies[key]))
```

Comment:
`serializeCookie` is not imported

Reply:
Oops, should be ok now :) Now I only need to look at the isUnset thing above.